server/methods/messageSearch.js

```diff
@@ -2,32 +2,40 @@ import s from 'underscore.string';
 
 Meteor.methods({
 	messageSearch(text, rid, limit) {
+		check(text, String);
+		check(rid, String);
+		check(limit, Match.Optional(Number));
+
+		// TODO: Evaluate why we are returning `users` and `channels`, as the only thing that gets set is the `messages`.
```

Comment:
Can someone answer this question?

Reply:
Seems like we shouldn't be... This looks kinda like maybe duplicated from spotlight?  As that's the only thing I can think of that returns all 3

Reply:
@graywolf336 It probably was a copy of what slack does at that time:
![image](https://user-images.githubusercontent.com/234261/33519696-36b689ee-d793-11e7-8283-25ef017713b0.png)

I don't know if that is what we want now, need to discuss.

Reply:
Isn't this essentially what the spotlight methods do?

Reply:
@geekgonecrazy Yes, but it was prior to the spotlight implementation